### PR TITLE
Splitup the parser

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

* Starts splitting up the parser so that it will be easier to reason about. 
* This is step one of probably 2 or 3. I'll queue up another PR tomorrow morning.
* I'd like to organize the new files into a *utils* and *queries* dir or something like that
* I'd like to move the tests and fixtures into this directory
* I'd also like to have a nice test index.html file for testing out the parser with the fixtures...

### CODE REVIEW

File: src/utils/parser/getSymbols.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  location: ASTLocation,
+  parameterNames?: string[]
+};
+
+export type SymbolDeclarations = {
+  functions: Array<SymbolDeclaration>,
+  variables: Array<SymbolDeclaration>
+};
+
+function getVariableNames(path): SymbolDeclaration[] {
```
Comment (wldcordeiro) [inline src/utils/parser/getSymbols.js]: If getFunctionNames gets moved, how come not this?

**Comment by jasonLaster** [comment]
It's going soon too :)

File: src/utils/parser/getAst.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      console.warn("parse failed", text);
+    }
+
+    ast = {};
+  }
+
+  return ast;
+}
+
+export default function getAst(sourceText: SourceText) {
```
Comment (ryanjduffy) [inline src/utils/parser/getAst.js]: Seems like `getAst` isn't the right module name/default export if you're going to standardize on using `traverseAst` to fetch the AST and handle it. Suggest making this a named export and make the module be `./ast.js`

Comment (jasonLaster) [inline src/utils/parser/getAst.js]: that's a good point

File: src/utils/parser/getFunctionName.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+export default function getFunctionName(path): string {
```
Comment (jasonLaster) [inline src/utils/parser/getFunctionName.js]: I like it being separate because infering names is a special use case

**Pull request merged**